lib/active_merchant/billing/gateways/monei.rb

```diff
@@ -225,6 +226,36 @@ def add_customer(xml, credit_card, options)
         end
       end
 
+      # Private : Convert ECI to ResultIndicator
+      # Possible ECI values:
+      # 02 or 05 - Fully Authenticated Transaction
+      # 00 or 07 - Non 3D Secure Transaction
+      # Possible ResultIndicator values:
+      # 01 = MASTER_3D_ATTEMPT
+      # 02 = MASTER_3D_SUCCESS
+      # 05 = VISA_3D_SUCCESS
+      # 06 = VISA_3D_ATTEMPT
+      # 07 = DEFAULT_E_COMMERCE
+      def eci_to_result_indicator(eci)
+        case eci
+        when '02', '05'
+          return eci
+        else
+          return '07'
+        end
+      end
+
+      # Private : Add the 3DSecure infos to XML
+      def add_three_d_secure(xml, options)
+        if options[:three_d_secure]
+          xml.Authentication(:type => '3DSecure') do
```

Comment:
is there a test for this?

Reply:
I might just be a question of re-recording the VCR cassettes

Reply:
@jimmyn Can you add those changes @elfassy suggested?

Reply:
@filipebarcos done

New remote test results:

```
rake test TEST=test/remote/gateways/remote_monei_test.rb
/Users/dmitriy/.rvm/rubies/ruby-2.5.5/bin/ruby -w -I"lib:test" -I"/Users/dmitriy/.rvm/gems/ruby-2.5.5/gems/rake-12.3.3/lib" "/Users/dmitriy/.rvm/gems/ruby-2.5.5/gems/rake-12.3.3/lib/rake/rake_test_loader.rb" "test/remote/gateways/remote_monei_test.rb" 
Loaded suite /Users/dmitriy/.rvm/gems/ruby-2.5.5/gems/rake-12.3.3/lib/rake/rake_test_loader
Started
..................
Finished in 14.039476 seconds.
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
18 tests, 42 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
1.28 tests/s, 2.99 assertions/s
```